processor/consent.go

```diff
@@ -47,28 +56,38 @@ func (proc *Handle) getConsentFilteredDestinations(event types.SingularEventT, s
 		proc.logger.Errorw("failed to get consent management info", "error", err.Error())
 	}
 
-	if len(consentManagementInfo.DeniedConsentIDs) == 0 {
+	// If the event does not have any consents, do not filter any destinations
+	if len(consentManagementInfo.AllowedConsentIDs) == 0 && len(consentManagementInfo.DeniedConsentIDs) == 0 {
 		return destinations
 	}
 
 	return lo.Filter(destinations, func(dest backendconfig.DestinationT, _ int) bool {
 		// Generic consent management
 		if cmpData := proc.getGCMData(sourceID, dest.ID, consentManagementInfo.Provider); len(cmpData.Consents) > 0 {
-
 			finalResolutionStrategy := consentManagementInfo.ResolutionStrategy
 
 			// For custom provider, the resolution strategy is to be picked from the destination config
 			if consentManagementInfo.Provider == "custom" {
 				finalResolutionStrategy = cmpData.ResolutionStrategy
 			}
 
+			// TODO: Remove "or" and "and" support once all the SDK clients stop sending it.
```

Comment:
Should we emit a stat so that we can monitor if SDKs stopped sending this?

Reply:
Yeah, that's a good idea. Let me check.

Reply:
Done.